constructor.sh

```diff
@@ -230,14 +230,17 @@ popd >/dev/null
 
 ##### DOWNLOAD SOFTWARE #####
 
-add-apt-repository ppa:webupd8team/java
-${ADMIN_TOOLS_DIR}/apt-repo add beats "https://artifacts.elastic.co/packages/6.x/apt stable main" https://artifacts.elastic.co/GPG-KEY-elasticsearch
-
 proxy_log "apt"
 export DEBIAN_FRONTEND=noninteractive
 apt-get update
 dpkg --configure -a
 apt-get -f install
+apt-get -y install software-properties-common # this provides add-apt-repository
+
+add-apt-repository ppa:webupd8team/java
+${ADMIN_TOOLS_DIR}/apt-repo add beats "https://artifacts.elastic.co/packages/6.x/apt stable main" https://artifacts.elastic.co/GPG-KEY-elasticsearch
+apt-get update # again
```

Comment:
why does it have to be done again ?

Reply:
Because we have to install add-apt-repository, then use add-apt-repository to add the PPA, then install the software from inside the PPA. Sucks, I know. :-(

Reply:
Tell you what, I'll wrap this in a test so that it only runs if absolutely necessary, and then I'll add software-properties-common to our list of standard software so that new installs won't require it.

Reply:
makes sense thanks for the explanation